pinot-minion/src/main/java/org/apache/pinot/minion/BaseMinionStarter.java

```diff
@@ -107,6 +112,9 @@ public void init(PinotConfiguration config)
     MinionTaskZkMetadataManager minionTaskZkMetadataManager = new MinionTaskZkMetadataManager(_helixManager);
     _taskExecutorFactoryRegistry = new TaskExecutorFactoryRegistry(minionTaskZkMetadataManager, _config);
     _eventObserverFactoryRegistry = new EventObserverFactoryRegistry(minionTaskZkMetadataManager);
+    _executorService =
```

Comment:
rename it as sth more specific to minion event observers. Plus, if this `_executorService` doesn't get called by other methods of this class, I suggest making it as a local variable.

Reply:
the name and creating newCachedThreadPool are to be consistent with the one created in other components like controller.

Reply:
Then I think we should consider renaming the one in controller as well in order to make it more specific. And suggest making it as a local variable if it's not used outside of this method.

Reply:
actually, I'd need the reference to stop it in the BaseMinionStarter.stop() method.

Reply:
since the executor is owned by the Starter class, not by EventObservers, so I'd like to keep its generic name. btw, it's called `_cleanupExecutor` once passed to EventObservers.